internal/command/jsonformat/renderer.go

```diff
@@ -240,6 +264,49 @@ func (r Renderer) RenderHumanPlan(plan Plan, mode plans.Mode, opts ...RendererOp
 	}
 }
 
+func (r Renderer) RenderLog(log *JSONLog) error {
+	switch log.Type {
+	case LogApplyStart, LogApplyComplete, LogRefreshStart:
+		msg := fmt.Sprintf("[bold]%s[reset]", log.Message)
+		r.Streams.Println(r.Colorize.Color(msg))
+
+	case LogDiagnostic:
+		diag := format.DiagnosticFromJSON(log.Diagnostic, r.Colorize, 78)
+		r.Streams.Print(diag)
+
+	case LogOutputs:
+		if len(log.Outputs) > 0 {
+			r.Streams.Println(r.Colorize.Color("[bold][green]Outputs:[reset]"))
+			for name, output := range log.Outputs {
+				change := differ.FromJsonOutput(output)
+				ctype, err := ctyjson.UnmarshalType(output.Type)
+				if err != nil {
+					return err
+				}
+
+				outputDiff := change.ComputeDiffForType(ctype)
+				outputStr := outputDiff.RenderHuman(0, computed.RenderHumanOpts{
+					Colorize:              r.Colorize,
+					ShowUnchangedChildren: true,
+				})
+
+				msg := fmt.Sprintf("%s = %s", name, outputStr)
+				r.Streams.Println(r.Colorize.Color(msg))
+			}
+		}
+
+	case LogChangeSummary:
+		// We will only render the apply change summary since the renderer
+		// generates a plan change summary for us
+		if !strings.Contains(log.Message, "Plan") {
+			msg := fmt.Sprintf("[bold][green]%s[reset]", log.Message)
+			r.Streams.Println("\n" + r.Colorize.Color(msg) + "\n")
+		}
+	}
```

Comment:
This switch statement has no `default` case, which means that anything not explicitly handled here will be dropped. This is not forwards-compatible with Terraform versions which emit new logs, as they won't be rendered if the client version is older. That was the purpose of the `@message` field: allowing older clients to have a fallback output for new information.

To correct this I think we'll need a stop-list of messages we _don't_ want to render, and a default case which just prints out the `@message` field for any unrecognized types.